src/dataframerow/dataframerow.jl

```diff
@@ -272,9 +272,17 @@ end
 
 @noinline pushhelper!(x, r) = push!(x, x[r])
 
-function Base.push!(df::DataFrame, dfr::DataFrameRow; columns::Symbol=:equal)
-    if !(columns in (:equal, :intersect))
-        throw(ArgumentError("`columns` keyword argument must be `:equal` or `:intersect`"))
+function Base.push!(df::DataFrame, dfr::DataFrameRow; cols::Symbol=:equal,
+                    columns::Union{Nothing,Symbol}=nothing)
```

Comment:
How about merging this definition with the one in the other file? As far as I can tell, DataFrameRow behaves like a Dict for everything that is needed. Thus we could merge it to the Union{AbstractDict, NamedTuple}

Reply:
There are two reasons we cannot do it:
1. `DataFrameRow` is not defined when abstractdataframe.jl is loaded
2. this method has a fast special case handling if `DataFrameRow` matches the data frame to which it is pushed.

Reply:
1) I guess that's why large codebases `include` everything into a single file. Would that help here too?
2) In that case I would detect whether the special case is happening and if so call another function, or just enter through the other function and fallback to the general implementation if the special case didn't occur.

Reply:
Well that would make things significantly more complex (and where would that function live?), and it's not so hard to keep these in sync manually.

Reply:
I would leave code refactoring for later. In general your point 1. will not solve the issue (we already do this). The problem is with the code sequence. What will solve it (and we should do it but in a separate PR) is to move definitions of all types to a separate file and load it first. I am postponing this (and other refactoring) to post 1.0 release - as with @nalimilan we have agreed some time ago to focus now on finishing API to make 1.0 release as soon as possible.

For the second point - given that `vcat` will not have the same signature as `push!` and `append!` I am not sure it is worth it. Again - even if we decide to do this I would focus now on API and functionality and do the cleanup after 1.0 release.

Reply:
Leaving this for a separate "refactoring" PR.